modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx

```diff
@@ -78,9 +79,25 @@ const Actions = () => {
 	] = useState({});
 	const [iconSymbol, setIconSymbol] = useState('bolt');
 
+	const request = new XMLHttpRequest();
+
+	request.open('GET', spritemap, false);
+
+	request.send();
+
+	const svgDoc = request.responseXML;
+
+	const availableIconSymbolElements = svgDoc?.querySelectorAll('symbol');
+
+	const symbols = Array.from(availableIconSymbolElements!).map((element) => ({
+		label: Liferay.Language.get(element.id),
```

Comment:
This is not going to work as `Liferay.Language.get()` requires a static literal string to be substituted by the language filter server side, it can't manage dynamic strings which value is known browser side.

I would say not all clay icon symbol ids are translated (i.e they are not present in `Language.properties` files)  though, nevertheless, I think icon selection can be done by just presenting the icon symbol id

Reply:
Yea I saw that the string wasn't being translated, even though those strings i checked existed in language props. But yea, I remember something about not being able to pass a variable as an argument.

Reply:
@kresimir-coko instead of pulling the spritemap, you honesty might just be better off creating a js file that exports a giant json object with each label and icon name. This way you could also just import it instead of having to fetch it.

```js
exports default {
    accessibility: Liferay.Language.get('accessibility'),
    'add-cell': Liferay.Language.get('add-cell'),
    'add-row': Liferay.Language.get('add-row')
    // etc...
}
```

The only downfall of this is icons could be out of sync with our spritemap... But this may be more straight forward.


Another thought, is localization for icon names really necessary? cc @dsanz

Reply:
> is localization for icon names really necessary?

From what I understood it is not.

What @dsanz said:
> I think icon selection can be done by just presenting the icon symbol id

Reply:
@bryceosterhaus I managed to make our fetch work, making it more straightforward.

> The only downfall of this is icons could be out of sync with our spritemap

I mean, that can be a big problem down the line, but it might allow us to enable users to provide their own spritemaps (which is a known hurdle for this screen). 

Although the fetch approach can enable users that, too I AFAIK (we can add a prop for a path to the spritemap and the code is essentially the same).

IDK, what are your thoughts @dsanz do we go fetch or local file?

Reply:
After some thoughts, I believe we have 2 separate things to solve here:

1. Knowing the full set of symbols
2. Getting the translations

About (1) that's a new need as usually the symbol one wants to use is known by the client code. Now, we need the full set to fill the drowpdown in. We have the icons URL, and most likely, it will have already been fetched previously by the control panel page, so Kreso's fetch will likely get the **admin icons** cached response. If the icons were inlined in the dom, we could avoid this step and query dom to get the list, but that is not the case in DXP. I just mean the same as this [stackoverflow question](https://stackoverflow.com/questions/69801088/load-an-svg-image-where-the-elements-become-part-of-the-dom). I'm afraid we can't avoid that fetch even if we don't want translations.

About (2), the main caveat I see is that not all symbols we have in clay exist in `Language.properties`, so the solution of sending all translations would produce a dropdown where some icons show a translated name, some shown the symbol id. I've done a quick check:

```bash
/src [master]$ for i in  $(ls ./modules/apps/frontend-theme/frontend-theme-classic/build/images/clay/); do echo "${i%.*} :"; cat ./modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties | grep ^${i%.*}= | wc -l; done | grep 0 | wc -l
301

/src [master]$ for i in  $(ls ./modules/apps/frontend-theme/frontend-theme-classic/build/images/clay/); do echo "${i%.*} :"; cat ./modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties | grep ^${i%.*}= | wc -l; done | grep 1 | wc -l
160

```
So we have 160 icons translated and 301 untranslated at this moment, that's ~33%

Therefore, it sounds reasonable to me if we defer the translation of icons and just show the symbols. In any case, because of the way we use the svgs in the page, there seems no access to the full list of symbols just by querying the dom

Reply:
👍 I like the idea of having a separate task/story/spike where we improve this, but for now it works as it does in the Figma so it meets the acceptance criteria